### ISSUE TITLE
restoring failed: admin login responded with 419

### ISSUE THREAD

**Issue by ixtora** [Rhymen/go-whatsapp#268]
I'm trying to restore the session and gives an error.
Any have the same problem?

**Comment by ski7777** [comment]
Yes. Same problem here. After starting a new session the new one can be
restored without any problem

ixtora <notifications@github.com> schrieb am Di., 7. Jan. 2020, 11:46:

**Comment by ixtora** [comment]
@ski7777 An error occurs frequently? Can I start a session without scanning the qr code?

**Comment by AlmogBaku** [comment]
@Rhymen @tulir any idea?

**Comment by turcol** [comment]
Looks like WhatsApp session expire in 30 days, after that it would give you the message:

`[0030] FATAL main:         Starting gateway failed: Bridge whatsapp.bridgenaeme failed to start: failed to restore session: admin login responded with 419
`

**Comment by AlmogBaku** [comment]
Does the session expire after 30 days of inactivity or anyway?

**Comment by turcol** [comment]
> Does the session expire after 30 days of inactivity or anyway?

I am not sure if with no activity the session expires sooner. In our case, we had full activity during the 30 days

**Comment by AlmogBaku** [comment]
So... how does Whatsapp desktop overcome it?

**Comment by tulir** [comment]
There's no hard limit, I've been logged in with the same session for over a year. WhatsApp probably just randomly kicks people out if they feel like it.

**Comment by AlmogBaku** [comment]
@tulir I suspects that new sessions are being purged due to inactivity

**Comment by ciroiriarte** [comment]
Is it possible that we're missing some kind of "heartbeat" messaging?. Looking at the "linked devices" view, I have one session per month:

![image](https://user-images.githubusercontent.com/1750260/127867287-d7de89bc-c833-44e4-ae5d-cf44590e4cdd.png)

That seems to never refresh after initial authentication. I'm using the library through Matterbridge

**Comment by AlmogBaku** [comment]
You need to restore your login token.... @ciroiriarte

**Comment by ciroiriarte** [comment]
I'm using this library inderectly through https://github.com/42wim/matterbridge, it works fine, but just for 30 days.

**Comment by ciroiriarte** [comment]
> You need to restore your login token.... @ciroiriarte

It seems the restore is attempted, but each month gives: 

"failed to restore serssion: admin login responded with 419"

And you have to reauth.
